### PR TITLE
add RSICV-ASM grammar

### PR DESCRIPTION
**Pull request by GOKINGD**
add RISCV-ASM grammar (Only part of the instruction set, not including compiler instructions)

### CODE REVIEW

**Comment by kaby76** [comment]
Please add a desc.xml so that it can be tested. E.g., https://github.com/antlr/grammars-v4/blob/master/asm/asm6502/desc.xml declares that it works for all targets, and requires Antlr 4.10 as you use the `caseInsensitive = true;` option.

File: asm/asmriscv/Riscv64G.g4 [hunk 1]
```diff
@@ -0,0 +1,578 @@
...
+   |'pmpaddr1'
+   |'pmpaddr2'
+   |'pmpaddr3'
+   |'pmpaddr4'
+   |'pmpaddr5' 
+   |'pmpaddr6'
+   |'pmpaddr7'
+   ;
+
+COMMENT
```
Comment (kaby76) [inline asm/asmriscv/Riscv64G.g4:551]: You cannot have a rule that uses a token that is marked HIDDEN (i.e., COMMENT, in rule `comment: COMMENT;`. The parser only "sees" DEFAULT-marked channel tokens.

**Comment by kaby76** [comment]
Please add the grammar to the parent pom.xml. You will also need to add the parent to the pom.xml for the grammar. See how it is done in the other asm grammars.

**Comment by kaby76** [comment]
Please remove the useless parentheses.
```
$ bash /c/Users/Kenne/Documents/GitHub/g4-scripts/find-useless-parentheses.sh Riscv64G.g4
CSharp 0 Riscv64G.g4 success 0.0711347
Found useless parentheses in grammars...
Riscv64G.g4:L137:    : OP EXTENSIONNAME (number | (number 'p' number))?
                                                  ^
Riscv64G.g4:L568:    : ('0x')? [0-9A-F]+
                       ^
01/22-16:27:00 ~/test/grammars-v4/asm/asmriscv
$
```

**Comment by kaby76** [comment]
Please correct the pom.xml. There are references to the 8086 grammar in this new grammar.
```
01/22-18:31:24 ~/test/grammars-v4/asm/asmriscv
$ mvn clean test
[INFO] Scanning for projects...
[INFO]
[INFO] ---------------------< org.antlr.grammars:asm8086 >---------------------
[INFO] Building ASM 8086 grammar 1.0-SNAPSHOT
[INFO] --------------------------------[ jar ]---------------------------------
[INFO]
[INFO] --- maven-clean-plugin:2.5:clean (default-clean) @ asm8086 ---
[INFO] Deleting C:\msys64\home\Kenne\test\grammars-v4\asm\asmriscv\target
[INFO]
[INFO] --- antlr4-maven-plugin:4.11.1:antlr4 (default) @ asm8086 ---
[INFO] No grammars to process
[INFO] ANTLR 4: Processing source directory C:\msys64\home\Kenne\test\grammars-v4\asm\asmriscv
[INFO]
[INFO] --- maven-resources-plugin:2.6:resources (default-resources) @ asm8086 ---
[INFO] Using 'UTF-8' encoding to copy filtered resources.
[INFO] skip non existing resourceDirectory C:\msys64\home\Kenne\test\grammars-v4\asm\asmriscv\src\main\resources
[INFO]
[INFO] --- maven-compiler-plugin:3.11.0:compile (default-compile) @ asm8086 ---
[INFO] Nothing to compile - all classes are up to date
[INFO]
[INFO] --- maven-resources-plugin:2.6:testResources (default-testResources) @ asm8086 ---
[INFO] Using 'UTF-8' encoding to copy filtered resources.
[INFO] skip non existing resourceDirectory C:\msys64\home\Kenne\test\grammars-v4\asm\asmriscv\src\test\resources
[INFO]
[INFO] --- maven-compiler-plugin:3.11.0:testCompile (default-testCompile) @ asm8086 ---
[INFO] No sources to compile
[INFO]
[INFO] --- maven-surefire-plugin:2.12.4:test (default-test) @ asm8086 ---
[INFO] No tests to run.
[INFO]
[INFO] --- antlr4test-maven-plugin:1.22:test (default) @ asm8086 ---
[INFO] Evaluating Scenario: Default Scenario
java.lang.ClassNotFoundException: asm8086Lexer
        at java.base/java.net.URLClassLoader.findClass(URLClassLoader.java:445)
        at java.base/java.lang.ClassLoader.loadClass(ClassLoader.java:588)
        at java.base/java.lang.ClassLoader.loadClass(ClassLoader.java:521)
        at com.khubla.antlr.antlr4test.ScenarioExecutor.testGrammar(ScenarioExecutor.java:110)
        at com.khubla.antlr.antlr4test.ScenarioExecutor.testGrammars(ScenarioExecutor.java:229)
        at com.khubla.antlr.antlr4test.GrammarTestMojo.testScenarios(GrammarTestMojo.java:360)
        at com.khubla.antlr.antlr4test.GrammarTestMojo.execute(GrammarTestMojo.java:179)
        at org.apache.maven.plugin.DefaultBuildPluginManager.executeMojo(DefaultBuildPluginManager.java:137)
        at org.apache.maven.lifecycle.internal.MojoExecutor.doExecute(MojoExecutor.java:301)
        at org.apache.maven.lifecycle.internal.MojoExecutor.execute(MojoExecutor.java:211)
        at org.apache.maven.lifecycle.internal.MojoExecutor.execute(MojoExecutor.java:165)
        at org.apache.maven.lifecycle.internal.MojoExecutor.execute(MojoExecutor.java:157)
        at org.apache.maven.lifecycle.internal.LifecycleModuleBuilder.buildProject(LifecycleModuleBuilder.java:121)
        at org.apache.maven.lifecycle.internal.LifecycleModuleBuilder.buildProject(LifecycleModuleBuilder.java:81)
        at org.apache.maven.lifecycle.internal.builder.singlethreaded.SingleThreadedBuilder.build(SingleThreadedBuilder.java:56)
        at org.apache.maven.lifecycle.internal.LifecycleStarter.execute(LifecycleStarter.java:127)
        at org.apache.maven.DefaultMaven.doExecute(DefaultMaven.java:294)
        at org.apache.maven.DefaultMaven.doExecute(DefaultMaven.java:192)
        at org.apache.maven.DefaultMaven.execute(DefaultMaven.java:105)
        at org.apache.maven.cli.MavenCli.execute(MavenCli.java:960)
        at org.apache.maven.cli.MavenCli.doMain(MavenCli.java:293)
        at org.apache.maven.cli.MavenCli.main(MavenCli.java:196)
        at java.base/jdk.internal.reflect.DirectMethodHandleAccessor.invoke(DirectMethodHandleAccessor.java:104)
        at java.base/java.lang.reflect.Method.invoke(Method.java:577)
        at org.codehaus.plexus.classworlds.launcher.Launcher.launchEnhanced(Launcher.java:282)
        at org.codehaus.plexus.classworlds.launcher.Launcher.launch(Launcher.java:225)
        at org.codehaus.plexus.classworlds.launcher.Launcher.mainWithExitCode(Launcher.java:406)
        at org.codehaus.plexus.classworlds.launcher.Launcher.main(Launcher.java:347)
[INFO] ------------------------------------------------------------------------
[INFO] BUILD FAILURE
[INFO] ------------------------------------------------------------------------
[INFO] Total time:  1.068 s
[INFO] Finished at: 2024-01-22T18:31:29-05:00
[INFO] ------------------------------------------------------------------------
[ERROR] Failed to execute goal com.khubla.antlr:antlr4test-maven-plugin:1.22:test (default) on project asm8086: Unable execute mojo: asm8086Lexer -> [Help 1]
[ERROR]
[ERROR] To see the full stack trace of the errors, re-run Maven with the -e switch.
[ERROR] Re-run Maven using the -X switch to enable full debug logging.
[ERROR]
[ERROR] For more information about the errors and possible solutions, please read the following articles:
[ERROR] [Help 1] http://cwiki.apache.org/confluence/display/MAVEN/MojoExecutionException
01/22-18:31:30 ~/test/grammars-v4/asm/asmriscv
$
```

Also, Hello.S does not parse.
```
$ ./bin/Debug/net7.0/Test.exe ../examples/Hello.S
line 2:9 token recognition error at: '"'
line 39:13 token recognition error at: '"'
CSharp 0 ../examples/Hello.S fail 0.0580714
Total Time: 0.1232851
01/22-18:32:54 ~/test/grammars-v4/asm/asmriscv/Generated-CSharp
$
```

**Comment by GOKINGD** [comment]
oh~ Thank you very much for your review, I will revise these issues later.

File: asm/asmriscv/Riscv64G.g4 (hunk 1 above)
Comment (GOKINGD) [reply asm/asmriscv/Riscv64G.g4]: Yes, I get your mean, so I removed the HIDDEN-marked  tokens in parser.

File: asm/asmRISCV/pom.xml [hunk 2]
```diff
@@ -0,0 +1,55 @@
+<project xmlns="http://maven.apache.org/POM/4.0.0" xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance"
+	xsi:schemaLocation="http://maven.apache.org/POM/4.0.0 http://maven.apache.org/xsd/maven-4.0.0.xsd">
+	<modelVersion>4.0.0</modelVersion>
+	<artifactId>asmRSICV</artifactId>
+	<packaging>jar</packaging>
+	<name>ASM RSICV grammar</name>
+	<parent>
+		<groupId>org.antlr.grammars</groupId>
+		<artifactId>asmparent</artifactId>
```
Comment (kaby76) [inline asm/asmRISCV/pom.xml:9]: This looks ok, but the *parent* pom.xml, the one at .../asm/pom.xml, does not contain a link to this pom.xml, .../asm/asmriscv/pom.xml. It needs to be added:
```
01/23-07:07:52 ~/test/grammars-v4/asm
$ mvn clean test
[INFO] Scanning for projects...
[INFO] ------------------------------------------------------------------------
[INFO] Reactor Build Order:
[INFO]
[INFO] Assembler Grammars                                                 [pom]
[INFO] ASM 6502 grammar                                                   [jar]
[INFO] ASM 8080 grammar                                                   [jar]
[INFO] ASM 8086 grammar                                                   [jar]
[INFO] MASM grammar                                                       [jar]
[INFO] ASM Z80 grammar                                                    [jar]
[INFO] MASM grammar                                                       [jar]
[INFO] NASM grammar                                                       [jar]
[INFO] ASM pdp7 grammar                                                   [jar]
[INFO] CUDA PTX ISA Grammars                                              [pom]
[INFO] CUDA PTX ISA 1.0 grammar                                           [jar]
[INFO] CUDA PTX ISA 2.1 grammar                                           [jar]
[INFO]
[INFO] --------------------< org.antlr.grammars:asmparent >--------------

...

[INFO] ------------------------------------------------------------------------
[INFO] Reactor Summary for Assembler Grammars 1.0-SNAPSHOT:
[INFO]
[INFO] Assembler Grammars ................................. SUCCESS [  0.341 s]
[INFO] ASM 6502 grammar ................................... SUCCESS [  6.616 s]
[INFO] ASM 8080 grammar ................................... SUCCESS [  0.878 s]
[INFO] ASM 8086 grammar ................................... SUCCESS [  1.592 s]
[INFO] MASM grammar ....................................... SUCCESS [  1.135 s]
[INFO] ASM Z80 grammar .................................... SUCCESS [  0.601 s]
[INFO] MASM grammar ....................................... SUCCESS [  0.730 s]
[INFO] NASM grammar ....................................... SUCCESS [  4.373 s]
[INFO] ASM pdp7 grammar ................................... SUCCESS [  3.268 s]
[INFO] CUDA PTX ISA Grammars .............................. SUCCESS [  0.002 s]
[INFO] CUDA PTX ISA 1.0 grammar ........................... SUCCESS [  1.161 s]
[INFO] CUDA PTX ISA 2.1 grammar ........................... SUCCESS [ 20.722 s]
[INFO] ------------------------------------------------------------------------
[INFO] BUILD SUCCESS
[INFO] ------------------------------------------------------------------------
[INFO] Total time:  41.661 s
[INFO] Finished at: 2024-01-23T07:08:42-05:00
[INFO] ------------------------------------------------------------------------
01/23-07:08:42 ~/test/grammars-v4/asm
$ ls
asm6502/  asm8080/  asm8086/  asmMASM/  asmriscv/  asmZ80/  masm/  nasm/  pdp7/  pom.xml  ptx/
01/23-07:09:06 ~/test/grammars-v4/asm
$ grep asmriscv pom.xml
01/23-07:09:15 ~/test/grammars-v4/asm
$
```

**Review by kaby76 — APPROVED** [review]

**Comment by teverett** [comment]
@GOKINGD  thanks!

**Pull request merged**